stylus-sdk/src/evm.rs

```diff
@@ -57,6 +57,14 @@ pub fn pay_for_memory_grow(pages: u16) {
     unsafe { hostio::memory_grow(pages) }
 }
 
+/// Exits program execution early with the given status code.
+/// If `0`, the program returns successfully with any data supplied by `write_result`.
+/// Otherwise, the program reverts and treats any `write_result` data as revert data.
+#[inline]
+pub fn exit_early(status: u32) {
+    unsafe { hostio::exit_early(status) }
+}
```

Comment:
Instead of exposing this as-is to users, let's add a `revert!` macro that expands to a `console!`, `write_result` and `early_exit()`. Also let's go through the SDK and find places where we can replace panics with it.

We'll also want an affordance for the success case, which should ideally take an arguement for the return data.

Reply:
I added this macro. Note that `early_exit` does need to still be public so that it can be called by the expanded macro. I also added a similar `succeed` macro.

Currently the macro does not call `write_result` - it seemed awkward to me that the macro would receive both a result slice for revert data and a message to print. The user can first call `output`. However, if you do prefer the macro handle both (presumably just called as `revert!(slice_data, "Some console message")`, let me know and I'll make that change.